src/main/java/com/google/opensesame/servlets/UserServlet.java

```diff
@@ -73,58 +58,50 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   // Send a user to datastore. Update the current information about the user if one exists.
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String userGitHubID = request.getParameter("githubID");
-    if (userGitHubID == null || userGitHubID.trim().length() == 0) {
+    String userGitHubAuthToken = request.getParameter("gitHubAuthToken");
+    if (userGitHubAuthToken == null) {
       ErrorResponse.sendJsonError(
           response,
-          "Empty GitHub ID",
-          HttpServletResponse.SC_BAD_REQUEST,
-          "There is no GitHub ID affiliated with this request");
+          "GitHub OAuth token was never supplied",
+          HttpServletResponse.SC_UNAUTHORIZED,
+          "User could not be authenticated by GitHub, please try again");
       return;
     }
-    String[] userTags = request.getParameterValues("tags");
-    ArrayList<String> userTagsList =
-        userTags == null ? new ArrayList<String>() : new ArrayList<>(Arrays.asList(userTags));
+    String[] interestTagsParam = request.getParameterValues("interestTags");
```

Comment:
nit: remove `param` because:
 1. it doesn't give any extra information about what this variable does
 2. IIUC the data stored in `interestTagsParam` is the value for the "interestTags" parameter, so if we want to keep the suffix (which suggest against), we should change `Param` to `Value`

Reply:
The main aim of this naming scheme was to differentiate between the ```String[]``` gotten from the request and the ```ArrayList``` used in building a ```UserEntity``` . but I see what you are saying. I'll avoid creating two variables entirely by casting to an ```ArrayList``` when I build the ```UserEntity```